operator/redisfailover/service/client.go

```diff
@@ -146,8 +146,11 @@ func (r *RedisFailoverKubeClient) ensurePodDisruptionBudget(rf *redisfailoverv1.
 	namespace := rf.Namespace
 
 	minAvailable := intstr.FromInt(2)
-	if rf.Spec.Redis.Replicas <= 2 {
-		minAvailable = intstr.FromInt(int(rf.Spec.Redis.Replicas - 1))
+	if rf.Spec.Redis.Replicas == 2 {
+		minAvailable = intstr.FromInt(1)
+	} else if rf.Spec.Redis.Replicas <= 1 {
+		// No need to create this resource
+		return nil
```

Comment:
```suggestion
	if rf.Spec.Redis.Replicas <= 2 {
		minAvailable = intstr.FromInt(1)
```

After rethinking on this piece of code I think the best approach would be having a PDB with minAvailable of 1 when you set your replicas number <= 2

In the case replicas = 0 I don't see any problem to have the PDB created